File: c_programming/factorial.c
#include <stdio.h>
#include <stdlib.h>

long long int fac(int v)
{
	if(v==1) return v;
	return v*fac(v-1);
}

int main()
{
	int num;
  printf("Enter the number whose factorial is to be calculated:\t");
	scanf("%d",&num);
	long long int fac_value;
	fac_value=fac(num);
  printf("Factorial value of %d is:\n",num);
	printf("%Ld\n",fac_value);
	
	return 0;
}

File: factorial/java/Factorial.java
import java.util.Scanner;
/**
 * Created by buddhimah on 10/16/2017.
 */
public class Factorial {
    int result[10000];
    public static  int fac(int number){
        if (number >= 0) {
          result[0] = 1;
          for (int i = 1; i <= n; ++i) {
             result[i] = i * result[i - 1];
          }
          return result[n];
        }
   }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a number: ");
        int num =sc.nextInt();
        if(num < 0){
            Error a = new Error("factorial for negative numbers is undefined");
            System.out.println(a.getMessage());
            Factorial.main(new String[1]);
        }else {
            System.out.println(fac(num));
        }
    }
}
